### ISSUE TITLE
Override dependency scan status when no packages are flagged as Used

### ISSUE THREAD

**Issue by erichs** [ShiftLeftSecurity/sast-scan#233]
It would be great if the Dependency scan would only fail if vulnerable packages were flagged as Used.

Currently, the tool provides a recommendation like:
✅  `No package requires immediate attention since the major vulnerabilities are found only in dev packages and indirect dependencies.`

However, the Scan Summaries contain a ❌   and the final RC (`$?`) value is 1.

![Screen Shot 2020-11-02 at 3 08 07 PM](https://user-images.githubusercontent.com/513523/97914337-b69e0b00-1d1d-11eb-9382-f37f82e1dbb8.png)

**Comment by erichs** [comment]
This occurs when run with the flags: `--mode ci --type credscan,nodejs,depscan`

**Comment by prabhu** [comment]
Hi @erichs 

Great feedback!

Only this weekend, we updated build breaker logic to include dependency scan results. The challenge we have got is that the usage detection support is limited to node.js and java+maven alone. So the default rule is unfortunately mixing and matching two kinds of rules here - https://github.com/ShiftLeftSecurity/sast-scan/blob/master/lib/config.py#L1247

- max_severity - Applicable for all projects
- max_required_severity - Applicable for projects with package usage detection

In this case, max_high > 2 is triggering the build breaker so it exits with 1. To fix this issue, we can enhance this line to calculate both aggregate and required packages metrics - https://github.com/ShiftLeftSecurity/sast-scan/blob/master/lib/analysis.py#L103 . We can then use either the required packages metrics (if available) or use the aggregate metrics here - https://github.com/ShiftLeftSecurity/sast-scan/blob/master/lib/analysis.py#L136
